### ISSUE TITLE
Client "Failed to take screenshot" 

### ISSUE THREAD

**Issue by britmob** [migueldemoura/myazo#17]
Getting "failed to take screenshot" after running the client script on windows 17134.471. I've double checked everything was configured properly. 

I tried changing "snippingtool" in the script to "snippingtool.exe", same issue, and then I tried changing it to something completely unrelated to snippingtool, same error. This leads me to believe it's an issue with calling the snipping tool. 

Any help would be greatly appreciated because I very much want to migrate away from gyazo! :)

**Comment by migueldemoura** [comment]
The error you received should have a status code: `Error: Failed to upload screenshot. Server returned status code %s`. Could you point out which one it is?

Also, what is your configuration (minus the secret and domain)? Are you using the gyazo endpoint or the myazo server script?

Thanks! :smile:

**Comment by britmob** [comment]
![cmd](https://user-images.githubusercontent.com/26635615/50122976-e7a9eb00-022c-11e9-81ca-51222e8cb060.png)


What I get when I run it in cmd, and my config. (I can't upload .ini files to github so I changed it to .txt)

[config.txt](https://github.com/migueldemoura/myazo/files/2688253/config.txt)

I am using the myazo server script.

**Comment by migueldemoura** [comment]
My bad, I misread your error message.

Can you try running `snippingtool /clip` directly in your cmd? Is the image copied to your clipboard?
There may have seem some changes to it, I don't have a windows box or VM right now.

**Comment by britmob** [comment]
Running snippingtool /clip puts me right at where I can select the area for a screenshot. Putting in the direct path also does the same thing.

**Comment by britmob** [comment]
I also tried running cmd in administrator mode with no luck.

**Comment by migueldemoura** [comment]
Can you try replacing this:

`myazo.py`
```python
for util, args in backends[platform.system()].items():
    if shutil.which(util) != None and run([util] + args).returncode == 0:
        break

# If the used util stored the screenshot in the clipboard, output it to the tmp file
if util == 'snippingtool':
    from PIL import ImageGrab
    img = ImageGrab.grabclipboard()
    if img is not None:
        img.save(tmp_file, optimize=True)

if os.path.isfile(tmp_file) != True:
    print('Error: Failed to take screenshot.')
    exit(-1)
```

with

`myazo.py`
```python
print(tmp_file)

for util, args in backends[platform.system()].items():
    if shutil.which(util) != None and run([util] + args).returncode == 0:
        print('Valid util: {}'.format(util))
        break

# If the used util stored the screenshot in the clipboard, output it to the tmp file
if util == 'snippingtool':
    print('Util is snippingtool')
    from PIL import ImageGrab
    img = ImageGrab.grabclipboard()
    if img is not None:
        print('Valid clipboard data')
        img.save(tmp_file, optimize=True)

if os.path.isfile(tmp_file) != True:
    print('Error: Failed to take screenshot.')
    exit(-1)
```

And run myazo from the cmd again. Running administrator 

I'll probably add some debug logging for these kinds of situations in the future.

**Comment by britmob** [comment]
![failed](https://user-images.githubusercontent.com/26635615/50188080-5a2ecf80-02ee-11e9-9b96-109782444b1f.png)

Sorry about the slow response.

**Comment by britmob** [comment]
Am able to take screenshot now, but getting a 404 error. 
![404](https://user-images.githubusercontent.com/26635615/50188217-d6291780-02ee-11e9-9bf0-0396d8e58dcc.png)

Here's a photo of my webserver directory. Using nginx for reference. 
![nginx](https://user-images.githubusercontent.com/26635615/50188271-0375c580-02ef-11e9-87cf-4f3fded3faac.png)

(I realize this is probably a stupid mistake on my part, almost a complete noob nginx-wise)

**Comment by migueldemoura** [comment]
Thanks a bunch for the info.

I hardcoded the slash `/` in the tmp filename, let's try and change that:

`myazo.py`
```python
tmp_file = '{}/{}.png'.format(tempfile.gettempdir(), next(tempfile._get_candidate_names()))
```

with

`myazo.py`
```python
tmp_file = os.path.abspath(
    '{}/{}.png'.format(tempfile.gettempdir(), next(tempfile._get_candidate_names()))
)
```

Also, can you change this:

`myazo.py`
```python
    if img is not None:
        img.save(tmp_file, optimize=True)
```
to

`myazo.py`
```python
    if img is not None:
        print(img.save(tmp_file, optimize=True))
```

Please check if the `tmp_file` exists and provide a screenshot/paste of the cmd output. Thanks! :smiley:

**Comment by britmob** [comment]
![failed](https://user-images.githubusercontent.com/26635615/50188925-4fc20500-02f1-11e9-8179-e0b604ece9f0.png)

Yes, C:\Users\(user)\Appdata\Local\temp\o_c0vca8.png exists.

**Comment by migueldemoura** [comment]
Great! The path was indeed an issue - I'll issue an patch once we figure out why you're getting the 404.

Can you validate that your `upload.php` script is reachable via a browser (the one you used in the `config.ini`)?

**Comment by britmob** [comment]
Trying to get nginx working. Works on index.html but won't respond on upload.php.

**Comment by migueldemoura** [comment]
Released [v0.6](https://github.com/migueldemoura/myazo/releases/tag/v0.6) which addresses the issue you found.

If you want, you can post your nginx config here and I can try to help.
